From d52361886f8f0da0d1cbbcba6431645a81c498d0 Mon Sep 17 00:00:00 2001
From: SangeethaRao <sangeetha.rao@lsi.com>
Date: Wed, 18 Sep 2013 02:09:31 -0500
Subject: [PATCH 65/75] arm/mach-axxia: Fixed type1 PCIe enumeration issue

PCI_FUNC() and PCI_SLOT() macros are already doing the masking and
shifting and it was incorrectly being done twice.

Signed-off-by: SangeethaRao <sangeetha.rao@lsi.com>
---
 arch/arm/mach-axxia/pci.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-axxia/pci.c b/arch/arm/mach-axxia/pci.c
index b0ddab6..2600892 100644
--- a/arch/arm/mach-axxia/pci.c
+++ b/arch/arm/mach-axxia/pci.c
@@ -209,8 +209,8 @@ axxia_pciex_get_config_base(struct axxia_pciex_port *port,
 		return port->regs;
 
 	relbus = bus->number - (port->root_bus_nr + 1);
-	dev    = (PCI_SLOT(devfn) & 0xf8) >> 3;
-	fn     = (PCI_FUNC(devfn) & 0x7);
+	dev    = PCI_SLOT(devfn);
+	fn     = PCI_FUNC(devfn);
 
 	if (dev > 31)
 		return NULL;
-- 
1.7.5.4

